tests/test_simple_pipeline_executor.py

```diff
@@ -243,14 +237,21 @@ def test_from_pipeline_output_differ(self):
             )
         )
 
+        executor = self._configure_pipeline(
+            NoDimensionsTestTask.ConfigClass,
+            NoDimensionsTestTask.ConfigClass,
+            storageClass_a="TaskMetadataLike",
+        )
+
         with self.assertLogs("lsst", level="INFO") as cm:
             quanta = executor.run(register_dataset_types=True, save_versions=False)
-        # a has been told to return a TaskMetadata but will convert to dict.
+        # a has been told to return a TaskMetadata but this will convert to
+        # dict on read by b.
         # b returns a dict and that is converted to TaskMetadata on put.
         self._test_logs(cm.output, "dict", "lsst.pipe.base.TaskMetadata", "dict", "dict")
 
         self.assertEqual(len(quanta), 2)
-        self.assertEqual(self.butler.get("intermediate"), {"zero": 0, "one": 1})
+        self.assertEqual(self.butler.get("intermediate").to_dict(), {"zero": 0, "one": 1})
         self.assertEqual(self.butler.get("output").to_dict(), {"zero": 0, "one": 1, "two": 2})
```

Comment:
So "output" is a TaskMetadata? How is that consistent with "b returned a dict" test above and the default connections being "dict"?

Reply:
Ok. It should likely use the more explicit TaskMetadata.from_dict() in the test to make it obvious.